addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -42,10 +42,11 @@ var LinkDialog = Dialog.extend({
         });
 
         this.data = linkInfo || {};
-        this.isButton = this.data.isButton;
+
+        const isHighlightedLink = this.data.range.isOnAnchor() && this.data.className.includes("btn-link");
```

Comment:
The check looks good to me, but I think the button logic might be intended and good to keep (that a real `<button/>` 
 should always have a btn look) and that we should have:

```suggestion
        const isButtonLink = this.isButton || (this.data.range.isOnAnchor() && this.data.className.includes("btn-link"));
```